tests/t525-operator.c

```diff
@@ -74,14 +82,20 @@ int main(int argc, char **argv) {
   CeedCompositeOperatorAddSub(op_composite, op_sub_2);
 
   // Check setting field in context of single sub-operator for composite operator
-  CeedOperatorContextSetDouble(op_composite, "time", 2.0);
+  CeedOperatorContextGetFieldLabel(op_composite, "time", &time_label);
+  double value_time = 2.0;
+  CeedOperatorContextSetDouble(op_composite, time_label, &value_time);
   if (ctx_data_2.time != 2.0)
     // LCOV_EXCL_START
     printf("Incorrect context data for time: %f != 2.0", ctx_data_2.time);
   // LCOV_EXCL_STOP
 
   // Check setting field in context of multiple sub-operators for composite operator
-  CeedOperatorContextSetDouble(op_composite, "other", 9000.);
+  CeedOperatorContextGetFieldLabel(op_composite, "other", &other_label);
+  // No issue requesting same label twice
+  CeedOperatorContextGetFieldLabel(op_composite, "other", &other_label);
```

Comment:
Looks good. How should we manage lifetimes for this concept in Rust? It's like an immutable borrow (because you need to call `CeedOperatorContextSetDouble` to mutate), but it's also safe to mutate the Operator while borrowed. But this label (maybe we should call it a Key) can't outlive the Operator. The GhostCell concept is similar, but we don't need its performance (lock elision).

Reply:
We can only express this concept in the Rust interface with gallery QFunctions since the context holds the user QFunction closure for non-gallery functions. I'm not sure how this will shape up once start adding in GPU capability or Gallery QFs with labeled fields. I think we can say that a derived object cannot outlive it's parent.